docs/standard/library-guidance/breaking-changes.md

```diff
@@ -0,0 +1,98 @@
+
+### Source breaking change
+
+A source breaking change doesn't affect program execution but will cause compilation errors the next time the application is recompiled. For example, a new overload can create an ambiguity in method calls that were unambiguous previously, or a renamed parameter will break callers that use named parameters.
+
+```csharp
+public class Task
+{
+    // Adding a type called Task could conflict with System.Threading.Tasks.Task at compilation
+}
+```
+
+Because a source breaking change is only harmful when the developer recompiles their application, it's the least disruptive breaking change. Developers can fix their own broken source code easily.
+
+### Behavior breaking change
+
+Behavior changes are the most common breaking change: almost any change in behavior could break someone. Changing the results a method produces, the exception your library throws, or the data output it writes to the network or disk could all negatively impact applications and libraries that depend on you. Even a bug fix can qualify as a breaking change if users relied on the previously broken behavior.
+
+Adding features and improving bad behaviors is a good thing, but without care it can make it very hard for existing software to upgrade. One approach to helping developers deal with behavior breaking changes is to hide them behind settings. Settings let developers update to the latest version of your library while at the same time choosing to opt in or opt out of breaking changes. This strategy lets developers stay up-to-date while letting their consuming code adapt over time.
+
+For example, ASP.NET Core MVC has the concept of a [compatibility version](/aspnet/core/mvc/compatibility-version) that modifies the features enabled and disabled on `MvcOptions`.
+
+**✔️ CONSIDER** leaving new features off by default if they affect existing users, and let developers opt in to the feature with a setting.
+
+### Binary breaking change
+
+A binary breaking change happens when you change the public API of a library so assemblies compiled against older versions are no longer able to call it. For example, changing a method's signature by adding a new parameter will cause assemblies compiled against the older version of the library to throw a <xref:System.MissingMethodException>.
+
+A binary breaking change can also break an **entire assembly**. Renaming an assembly with `AssemblyName` will change the assembly's identity, as will adding, removing, or changing an assembly's strong naming key. A change of an assembly's identity will break all compiled code that uses it.
+
+**❌ DO NOT** change an assembly name.
+
+**❌ DO NOT** add, remove, or change the strong naming key.
+
+**✔️ CONSIDER** using abstract base classes instead of interfaces.
+
+> Adding anything to an interface will cause existing types that implement it to fail. An abstract base class allows you to add a default virtual implementation.
```

Comment:
This seems almost contradictory to the recommendations above: 

> Behavior changes are the most common breaking change: **almost any change in behavior could break someone**.

Modifying a base class, even if it is an addition of a virtual member, can be a breaking change. An author of a shared public API can never know or assume to know how the API is used in the wild.

This also goes against SOLID principles.

Reply:
Never changing isn't an option. You will always break people. The aim is to add the most value with the least disruption.